[2024-04-26T17:27:09.437+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-26T17:27:09.461+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-02T22:00:00+00:00 [queued]>
[2024-04-26T17:27:09.466+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-02T22:00:00+00:00 [queued]>
[2024-04-26T17:27:09.466+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-26T17:27:09.474+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): send_air_quality_alerts> on 2024-04-02 22:00:00+00:00
[2024-04-26T17:27:09.483+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=942) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-04-26T17:27:09.484+0000] {standard_task_runner.py:63} INFO - Started process 943 to run task
[2024-04-26T17:27:09.483+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'air_quality_pipeline', 'send_air_quality_alerts', 'backfill__2024-04-02T22:00:00+00:00', '--job-id', '361', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/air_quality/air_quality_dag.py', '--cfg-path', '/tmp/tmpsqo7dqbw']
[2024-04-26T17:27:09.485+0000] {standard_task_runner.py:91} INFO - Job 361: Subtask send_air_quality_alerts
[2024-04-26T17:27:09.520+0000] {task_command.py:426} INFO - Running <TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-02T22:00:00+00:00 [running]> on host 4718ecf2379c
[2024-04-26T17:27:09.577+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='air_quality_pipeline' AIRFLOW_CTX_TASK_ID='send_air_quality_alerts' AIRFLOW_CTX_EXECUTION_DATE='2024-04-02T22:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='backfill__2024-04-02T22:00:00+00:00'
[2024-04-26T17:27:09.578+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-26T17:27:09.587+0000] {tasks.py:94} INFO - Generating air quality alerts...
[2024-04-26T17:27:09.588+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-26T17:27:09.588+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/air_quality/tasks.py", line 95, in send_air_quality_alerts
    alert_message = air_quality_alerts()
                    ^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/src/air_quality_alerts.py", line 68, in air_quality_alerts
    data = pd.read_csv("air_quality_data.csv")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 948, in read_csv
    return _read(filepath_or_buffer, kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 611, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1448, in __init__
    self._engine = self._make_engine(f, self.engine)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1705, in _make_engine
    self.handles = get_handle(
                   ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/common.py", line 863, in get_handle
    handle = open(
             ^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'air_quality_data.csv'
[2024-04-26T17:27:09.596+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=air_quality_pipeline, task_id=send_air_quality_alerts, execution_date=20240402T220000, start_date=20240426T172709, end_date=20240426T172709
[2024-04-26T17:27:09.604+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 361 for task send_air_quality_alerts ([Errno 2] No such file or directory: 'air_quality_data.csv'; 943)
[2024-04-26T17:27:09.617+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-04-26T17:27:09.631+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-26T17:27:09.632+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-04-26T17:32:56.699+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-26T17:32:56.725+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-02T22:00:00+00:00 [queued]>
[2024-04-26T17:32:56.729+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-02T22:00:00+00:00 [queued]>
[2024-04-26T17:32:56.730+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-26T17:32:56.737+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): send_air_quality_alerts> on 2024-04-02 22:00:00+00:00
[2024-04-26T17:32:56.745+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1263) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-04-26T17:32:56.746+0000] {standard_task_runner.py:63} INFO - Started process 1264 to run task
[2024-04-26T17:32:56.745+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'air_quality_pipeline', 'send_air_quality_alerts', 'backfill__2024-04-02T22:00:00+00:00', '--job-id', '381', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/air_quality/air_quality_dag.py', '--cfg-path', '/tmp/tmp3ckcpjsd']
[2024-04-26T17:32:56.746+0000] {standard_task_runner.py:91} INFO - Job 381: Subtask send_air_quality_alerts
[2024-04-26T17:32:56.779+0000] {task_command.py:426} INFO - Running <TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-02T22:00:00+00:00 [running]> on host 4718ecf2379c
[2024-04-26T17:32:56.842+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='air_quality_pipeline' AIRFLOW_CTX_TASK_ID='send_air_quality_alerts' AIRFLOW_CTX_EXECUTION_DATE='2024-04-02T22:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='backfill__2024-04-02T22:00:00+00:00'
[2024-04-26T17:32:56.843+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-26T17:32:56.852+0000] {tasks.py:94} INFO - Generating air quality alerts...
[2024-04-26T17:32:56.861+0000] {tasks.py:102} INFO - Cleaning up temporary files...
[2024-04-26T17:32:56.861+0000] {tasks.py:105} INFO - Temporary files cleaned up.
[2024-04-26T17:32:56.861+0000] {python.py:237} INFO - Done. Returned value was: None
[2024-04-26T17:32:56.862+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-26T17:32:56.868+0000] {taskinstance.py:1205} INFO - Marking task as SUCCESS. dag_id=air_quality_pipeline, task_id=send_air_quality_alerts, execution_date=20240402T220000, start_date=20240426T173256, end_date=20240426T173256
[2024-04-26T17:32:56.919+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2024-04-26T17:32:56.933+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-26T17:32:56.936+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-04-26T17:41:22.976+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-26T17:41:23.000+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-02T22:00:00+00:00 [queued]>
[2024-04-26T17:41:23.005+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-02T22:00:00+00:00 [queued]>
[2024-04-26T17:41:23.005+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-26T17:41:23.013+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): send_air_quality_alerts> on 2024-04-02 22:00:00+00:00
[2024-04-26T17:41:23.022+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1597) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-04-26T17:41:23.023+0000] {standard_task_runner.py:63} INFO - Started process 1598 to run task
[2024-04-26T17:41:23.022+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'air_quality_pipeline', 'send_air_quality_alerts', 'backfill__2024-04-02T22:00:00+00:00', '--job-id', '398', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/air_quality/air_quality_dag.py', '--cfg-path', '/tmp/tmpf1ymjphq']
[2024-04-26T17:41:23.023+0000] {standard_task_runner.py:91} INFO - Job 398: Subtask send_air_quality_alerts
[2024-04-26T17:41:23.055+0000] {task_command.py:426} INFO - Running <TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-02T22:00:00+00:00 [running]> on host 4718ecf2379c
[2024-04-26T17:41:23.119+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='air_quality_pipeline' AIRFLOW_CTX_TASK_ID='send_air_quality_alerts' AIRFLOW_CTX_EXECUTION_DATE='2024-04-02T22:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='backfill__2024-04-02T22:00:00+00:00'
[2024-04-26T17:41:23.120+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-26T17:41:23.128+0000] {tasks.py:94} INFO - Generating air quality alerts...
[2024-04-26T17:41:23.136+0000] {tasks.py:102} INFO - Cleaning up temporary files...
[2024-04-26T17:41:23.136+0000] {tasks.py:105} INFO - Temporary files cleaned up.
[2024-04-26T17:41:23.136+0000] {python.py:237} INFO - Done. Returned value was: None
[2024-04-26T17:41:23.136+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-26T17:41:23.143+0000] {taskinstance.py:1205} INFO - Marking task as SUCCESS. dag_id=air_quality_pipeline, task_id=send_air_quality_alerts, execution_date=20240402T220000, start_date=20240426T174123, end_date=20240426T174123
[2024-04-26T17:41:23.195+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2024-04-26T17:41:23.210+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-26T17:41:23.212+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
